app/commands/user/reset_cache.rb

```diff
@@ -24,4 +27,11 @@ def cache
   def value_for_has_unrevealed_testimonials? = user.mentor_testimonials.unrevealed.exists?
   def value_for_has_unrevealed_badges? = user.acquired_badges.unrevealed.exists?
   def value_for_has_unseen_reputation_tokens? = user.reputation_tokens.unseen.exists?
+  def value_for_num_solutions_mentored = user.mentor_discussions.count
+  def value_for_num_testimonials = user.mentor_testimonials.published.count
+
+  # This one is sloooooow!
```

Comment:
Maybe we could join on request instead of solution? Might be faster

Reply:
I considered that too. But discovered that `request` would have to join to `solution` to get the `user_id`.

Reply:
But couldn't we use the `student_id` field in `Mentor::Request`?

Reply:
I looked into this a bit, and found something I don't understood. How could there be a mentoring request for a solution where the request's student is different from the solution's user?

Reply:
Joining on request would be way faster (mentor ID redacted):

```
irb(main):028:0> user.mentor_discussions.joins(:request).distinct.count(:student_id)
D, [2023-05-18T12:06:37.624318 #71023] DEBUG -- :   Mentor::Discussion Count (4281.5ms)  SELECT COUNT(DISTINCT student_id) FROM `mentor_discussions` INNER JOIN `mentor_requests` ON `mentor_requests`.`id` = `mentor_discussions`.`request_id` WHERE `mentor_discussions`.`mentor_id` = X
=> 7642
irb(main):029:0> user.mentor_discussions.joins(:solution).distinct.count(:user_id)
D, [2023-05-18T12:07:13.302365 #71023] DEBUG -- :   Mentor::Discussion Count (19616.3ms)  SELECT COUNT(DISTINCT user_id) FROM `mentor_discussions` INNER JOIN `solutions` ON `solutions`.`id` = `mentor_discussions`.`solution_id` WHERE `mentor_discussions`.`mentor_id` = X
=> 7526
```